### PR TITLE
Improvement/update ruby version and fix parsed response

### PR DESCRIPTION
**Pull request by viktorianer**
Update gems, ruby version and more

### CODE REVIEW

File: lib/dailymotion-api/version.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
-# -*- encoding: utf-8 -*-
+# frozen_string_literal: true
 
 module DailymotionApi
-  VERSION = "0.2.1"
+  VERSION = "0.2.4"
```
Comment (ggarnier) [inline lib/dailymotion-api/version.rb:4]: Why did you skip versions `0.2.2` and `0.2.3`?

File: lib/dailymotion-api/client.rb [hunk 2]
```diff
@@ -46,31 +55,35 @@ def get_video(video_id, fields = "")
...
       response.parsed_response
     end
 
     def video_url
-      @video_url ||= get_video(@video_id, "url")["url"] rescue nil
+      @video_url ||=  begin
+                        get_video(@video_id, "url")["url"]
+                      rescue StandardError
+                        nil
+                      end
```
Comment (ggarnier) [inline lib/dailymotion-api/client.rb:69]: What's the purpose of this change? I think the original version is more concise

File: lib/dailymotion-api/client.rb [hunk 3]
```diff
@@ -46,31 +55,35 @@ def get_video(video_id, fields = "")
       return nil unless video_id
 
       url = "#{API_URL}/video/#{video_id}"
-      url << "?fields=#{fields}" unless fields.empty?
+      url += "?fields=#{fields}" unless fields.empty?
```
Comment (ggarnier) [inline lib/dailymotion-api/client.rb:58]: I don't see the point of this change. `url << "foo"` appends `"foo"` string to `url` object, and `url += "foo"` builds a new object:

```ruby
irb(main):001:0> url = ""
=> ""
irb(main):002:0> url.object_id
=> 46945403679320
irb(main):003:0> url << "foo"
=> "foo"
irb(main):004:0> url.object_id
=> 46945403679320
irb(main):005:0> url += "foo"
=> "foofoo"
irb(main):006:0> url.object_id
=> 46945403766300
```

**Review by ggarnier — COMMENTED** [review]
Hey @viktorianer, I'm sorry for such a long delay to review this. I did start reviewing a couple of weeks ago, but totally forgot I didn't finish it :disappointed: 

Anyway, here are a few comments. I didn't test it with a real video, though.

**Comment by viktorianer** [comment]
> Hey @viktorianer, I'm sorry for such a long delay to review this. I did start reviewing a couple of weeks ago, but totally forgot I didn't finish it 😞
> 
> Anyway, here are a few comments. I didn't test it with a real video, though.

No Problem. We have already decided to implement only a few endpoints for DM we need in our project. My changes were not enough, to make this gem usable for us. For example, the access token get expired, and we have to renew it.

Anyway, you can use my changes in our code base. I already removed the fork and my branch.

File: lib/dailymotion-api/client.rb (hunk 3 above)
Comment (viktorianer) [reply lib/dailymotion-api/client.rb:58]: Because `frozen_string_literal` does not allow you to do so any more. And yes, you could avoid creating 2 String objects, but it is not better, if you do so only ones.

https://docs.rubocop.org/rubocop-performance/cops_performance.html#performanceunfreezestring

File: lib/dailymotion-api/client.rb (hunk 2 above)
Comment (viktorianer) [reply lib/dailymotion-api/client.rb:69]: Because you hide all errors and don't catch only what you expect. I had no time to think about what errors you expect here. So, I just fixed Rubocop offence on this line.

File: lib/dailymotion-api/version.rb (hunk 1 above)
Comment (viktorianer) [reply lib/dailymotion-api/version.rb:4]: I found forks and PR with versions `0.2.2` and `0.2.3`. I thought that they might be merged before my PR :)
